news/utils.py

```diff
@@ -0,0 +1,138 @@
+    elif html_path is not None:
+        with open(html_path, "rb") as fp:
+            soup = BeautifulSoup(fp, "html.parser")
+
+    text = [soup.title.get_text() if soup.title else ""]
+
+    # Search for author
+    if byline := soup.find(class_=re.compile("byline")):
+        text.append(byline.get_text().strip())
+    elif byline := soup.find(class_=re.compile("post-author")):
+        text.append(byline.get_text().strip())
+    elif byline := soup.find(class_=re.compile("posted-by")):
+        text.append(byline.get_text().strip())
+    elif byline := soup.find(class_=re.compile("article__source")):
+        text.append(byline.get_text().strip())
+    elif byline := soup.find(class_=re.compile("author")):
+        text.append(byline.get_text().strip())
+
+    # Search for dateline
+    if dateline := soup.find("time", class_=re.compile("title")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("time", class_=re.compile("entry-date")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("time", class_=re.compile("date")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("time", class_="timestamp"):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("time", class_=re.compile("time")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("div", class_="timestamps"):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("div", class_="post-date"):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("div", class_="date"):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("span", class_=re.compile("date")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("span", class_=re.compile("posted-on")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find(class_=re.compile("article__date")):
+        text.append(dateline.get_text().strip())
+    elif dateline := soup.find("time"):
+        text.append(dateline.get_text().strip())
+
+    # article = soup.find_all(tag, attrs=attrs)
+    article = soup.find_all(tag, attrs={k:re.compile(v) for k,v in attrs.items()})
+    for a in article:
+
+        # Adapted from 
+        # https://github.com/bltlab/mot/blob/63ef942f2a4cc7fff5823b4cdefbccc5c7464b5f/extraction/extracttext.py#L540-L558
+        p_tag = a.find_all("p")
+        for p in p_tag:
+            # split_p = p.get_text().split("\n")
+            split_p = []
+            text_pieces = []
+            for child in p.children:
+                if type(child) is NavigableString:
+                    text_pieces.extend(child.split("\n"))
+                elif child.name == "br":
+                    split_p.append("".join(text_pieces))
+                    text_pieces = []
+                elif child.name == "em":
```

Comment:
Can the rest of this conditional can be simplified to something like the following?

```python
...
elif child.name in ("em", "a", ...):
  text_pieces.extend(child.get_text())
```

And maybe the `in` check should be against a variable like `FORMATTED_STRING_TAGS` or something?